### PR TITLE
Add nn graph doc

### PR DESCRIPTION
**Pull request by mir-of**

### CODE REVIEW

File: cn/docs/basics/08_nn_graph.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        # 注册模型 Module
+        self.model = nn.Linear(3, 1)
+
+        # 注册Loss
+        self.loss = nn.CrossEntropyLoss()  
+       
+        # 注册并配置 Optimzier
+        sgd_opt = optim.SGD(model1.conv1.weight, lr=<required parameter>, momentum=0)
+        lr_scheduler = optim.lr_scheduler.xxLR(sgd_opt)
+        self.add_optimizer(optimizer = sgd_opt,
```
Comment (strint) [inline cn/docs/basics/08_nn_graph.md:152]: 这一段还不可执行

File: cn/docs/basics/08_nn_graph.md [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        # 注册并配置 Optimzier
+        sgd_opt = optim.SGD(model1.conv1.weight, lr=<required parameter>, momentum=0)
+        lr_scheduler = optim.lr_scheduler.xxLR(sgd_opt)
+        self.add_optimizer(optimizer = sgd_opt, 
+                     lr_scheduler = lr_scheduler
+                     grad_clipping=ClipConf
+                     weight_decay=)
+  
+        # Graph级配置
+        self.train(True)
```
Comment (strint) [inline cn/docs/basics/08_nn_graph.md:158]: 这个原来在设计文档里面，现在已经废弃，通过是否有optimizer来判断是否是train

File: cn/docs/basics/08_nn_graph.md [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    f'Result: y = {linear_layer.bias.numpy()} + {linear_layer.weight[:, 0].numpy()} x + {linear_layer.weight[:, 1].numpy()} x^2 + {linear_layer.weight[:, 2].numpy()} x^3')
+```
+
+
+
+
+
+## OneFlow的Graph模式
+
+### `nn.graph`
```
Comment (strint) [inline cn/docs/basics/08_nn_graph.md:134]: nn.Graph

File: cn/docs/basics/08_nn_graph.md [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+目前，深度学习框架中模型的运行方式主要有两种，一种是以**动态图**的方式运行，即另一种是以**静态图**的方式运行。这两种方式各有优缺点，OneFlow对两种方式均提供了支持。
+
+
+
+在OneFlow中，以动态图运行的方式被称为***Eager模式***，以静态图运行的方式被称为***Graph模式***。
+
+
+
+其中，`nn.graph`为**静态图**运行方式的基础，本文将先介绍与**动态图/静态图**的概念，然后将具体介绍如何通过`nn.graph`，将OneFlow的模型的运行方式从**Eager模式**转换为**Graph模式**（即从动态图运行转换为静态图运行）。
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:11]: 之前大家达成一致，在 oneflow-documentation 里，中英文中间保留空格
```suggestion
其中，`nn.graph` 为**静态图**运行方式的基础，本文将先介绍与 **动态图/静态图** 的概念，然后将具体介绍如何通过`nn.graph`，将 OneFlow 的模型的运行方式从 **Eager 模式** 转换为 **Graph 模式**（即从动态图运行转换为静态图运行）。
```

另外，想看到编译的效果，可以安装仓库根目录下的 `requirement.txt`
```bash
pip install -r requirement.txt
```

然后进入到 `cn` 路径下编译：
```bash
cd cn
mkdocs build
```

File: cn/docs/basics/08_nn_graph.md [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+linear_t_g.debug()
+```
+
+输出中将显示出静态图的相关信息，包括计算图中各个节点的名称，层级关系，输入输出等信息。
+
+
+
+## 相关链接
+
+OneFlow Eager模式下的神经网络搭建：[搭建神经网络](https://github.com/Oneflow-Inc/oneflow-documentation/blob/new_interface_leap/cn/docs/basics/04_build_network.md)
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:257]: 同为一个 mkdocs 工程的文件（同cn目录下），可以用相对引用的，mkdocs 会自己最终替换为对应的 html 文件。
```suggestion
OneFlow Eager模式下的神经网络搭建：[搭建神经网络](./04_build_network.md)
```

File: cn/docs/basics/08_nn_graph.md [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+其中，`nn.graph`为**静态图**运行方式的基础，本文将先介绍与**动态图/静态图**的概念，然后将具体介绍如何通过`nn.graph`，将OneFlow的模型的运行方式从**Eager模式**转换为**Graph模式**（即从动态图运行转换为静态图运行）。
+
+
+
+## 动态图 vs. 静态图
+
+**动态图（Dynamic Graph）**
+
+* 动态图中的计算节点以*边定义边执行的方式运行（Define-by-Run）*，即计算节点的声明和真实赋值是同时进行的，因此也被称为*Eager模式/命令式编程 (Imperative)*。
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:19]: 这种 `*` 的 item list，需要多空一行，否则格式会出错。改成 `-` 就不用多空一行

File: cn/docs/basics/08_nn_graph.md [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+# tensor (x, x^2, x^3).
+p = flow.tensor([1, 2, 3], device=device, dtype=dtype)
+xx = x.unsqueeze(-1).pow(p)
+
+# The Linear Module
+model = flow.nn.Sequential(
+    flow.nn.Linear(3, 1),
+    flow.nn.Flatten(0, 1)
+)
+model = model.to(device)
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:93]: 这份代码有跑通过吗（除了修正没有对齐的pow外……）
```suggestion
model.to(device)
```
model 直接 to 即可。我试了按现在的写法，会在调用 model 算前向时报错。

另外，改了这个地方后，训练时跑飞了

File: cn/docs/basics/08_nn_graph.md [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+```python
+import math
+import numpy as np
+import oneflow as flow
+
+device = flow.device("cuda")
+dtype = flow.float32
+
+# Create Tensors to hold input and outputs.
+x = flow.tensor(np.linspace(-math.pi, math.pi, 20),
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:78]: pytorch 的版本是 2000

File: cn/docs/basics/08_nn_graph.md [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+# The Linear Module
+model = flow.nn.Sequential(
+    flow.nn.Linear(3, 1),
+    flow.nn.Flatten(0, 1)
+)
+model = model.to(device)
+
+# Loss Function
+loss_fn = flow.nn.MSELoss(reduction='sum')
+loss_fn = model.to(device)
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:97]:
```suggestion
model.to(device)
```

File: cn/docs/basics/08_nn_graph.md [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+
+device = flow.device("cuda")
+dtype = flow.float32
+
+# Create Tensors to hold input and outputs.
+x = flow.tensor(np.linspace(-math.pi, math.pi, 20),
+                device=device, dtype=dtype)
+y = flow.tensor(np.sin(x), device=device, dtype=dtype)
+
+# For this example, the output y is a linear function of (x, x^2, x^3), so
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:82]: 代码注释可以用中文。因为我们也有英文版的文档要做……

File: cn/docs/basics/08_nn_graph.md (hunk 9 above)
Comment (mir-of) [reply cn/docs/basics/08_nn_graph.md]: 这里有个小错误，应该是`loss_fn.to(device)`，这个改了后可以跑通

File: cn/docs/basics/08_nn_graph.md [hunk 11]
```diff
@@ -0,0 +1,336 @@
...
+| **计算方式** | Eager 模式                             | Lazy 模式                 |
+| **编程范式** | 命令式编程 (Imperative)               | 声明式编程 (Declarative) |
+| **优点**     | 代码编写灵活，易于调试                | 性能好，易于优化和部署   |
+| **缺点**     | 性能及可移植性差                      | 不易调试                 |
+
+
+
+## OneFlow 的 Eager 模式
+
+OneFlow默认以 Eager 模式运行，在[搭建神经网络](./04_build_network.md)的教程中，使用 [nn](https://oneflow.readthedocs.io/en/master/nn.html) 名称空间下的 API 搭建的网络，若不做特殊处理，均默认以 Eager 模式运行。
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:98]: 其实不止 `nn` 下的 API，还有 `flow.xxx` 以及 `flow.nn.functional` `flow.image` 等全都是默认eager。不过这好像比较麻烦描述，我想下怎么改。

File: cn/docs/basics/08_nn_graph.md [hunk 12]
```diff
@@ -0,0 +1,331 @@
+# 静态图模块 nn.Graph
+
+目前，深度学习框架中模型的运行方式主要有两种，即 **动态图** 与 **静态图**，在 OneFlow 中，也被习惯称为 **Eager 模式** 和 **Graph 模式** 。
+
+这两种方式各有优缺点，OneFlow 对两种方式均提供了支持，默认情况下是 Eager 模式。如果你是按顺序阅读本基础专题的教程，那么，到目前为所接触的所有代码都是 Eager 模式的代码。
```
Comment (yuanms2) [inline cn/docs/basics/08_nn_graph.md:5]: “到目前为”   --->  “到目前为止”
